crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -194,32 +232,22 @@ where
     Metadata: MetadataStorage,
     DA: DaBlockCostsSource,
 {
-    async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
-        let should_continue;
+    async fn run(&mut self, watcher: &mut StateWatcher) -> TaskRunResult {
         tokio::select! {
             biased;
             _ = watcher.while_started() => {
                 tracing::debug!("Stopping gas price service");
-                should_continue = false;
+                TaskRunResult::Stop
             }
             l2_block_res = self.l2_block_source.get_l2_block() => {
-                tracing::info!("Received L2 block result: {:?}", l2_block_res);
-                let block = l2_block_res?;
-
-                tracing::debug!("Updating gas price algorithm");
-                self.apply_block_info_to_gas_algorithm(block).await?;
-                should_continue = true;
+                let res = self.process_l2_block_res(l2_block_res).await;
+                TaskRunResult::continue_if_ok(res)
             }
             da_block_costs = self.da_source_channel.recv() => {
-                tracing::info!("Received DA block costs: {:?}", da_block_costs);
-                let da_block_costs = da_block_costs?;
-
-                tracing::debug!("Updating DA block costs");
-                self.apply_da_block_costs_to_gas_algorithm(da_block_costs).await?;
-                should_continue = true;
+                let res = self.process_da_block_costs_res(da_block_costs).await;
+                TaskRunResult::Continue
```

Comment:
If we have error here, you will not propagate it to the caller of the `run` function and we will miss log for this error.

Reply:
Thank you for catching this.